src/expressions/chainExpression.ts

```diff
@@ -126,43 +126,117 @@ module Plywood {
     }
 
     public foldIntoExternal(): Expression {
-      var externalExpression = this.expression;
-      var actions = this.actions;
+      var { expression, actions } = this;
+      var baseExternals = this.getBaseExternals();
+      if (baseExternals.length === 0) return this;
 
-      if (externalExpression instanceof ExternalExpression) {
+      // Looks like: External().blah().blah().blah()
+      if (expression instanceof ExternalExpression) {
         var undigestedActions: Action[] = [];
         for (var action of actions) {
-          var newExternal = (<ExternalExpression>externalExpression).addAction(action);
+          var newExternal = (<ExternalExpression>expression).addAction(action);
```

Comment:
we are only in here if the expression is an instanceof ExternalExpression right? (generics?)